server/models/profile.js

```diff
@@ -1,22 +1,97 @@
-module.exports = function(sequelize, DataTypes) {
-    var Profile = sequelize.define("Profile", {
-      firstName: DataTypes.STRING,
-      lastName: DataTypes.STRING,
-      city: DataTypes.STRING,
-      state: DataTypes.STRING,
-      age: DataTypes.INTEGER,
-      gender: DataTypes.STRING,
-      gitHub: DataTypes.STRING,
-      linkedIn: DataTypes.STRING,
-
-
-      
+module.exports = function(sequelize, DataTypes) 
+{
+    const profile = sequelize.define("profile",
+    {
+      id:
+      {
+          //there may be an issue if the column is not CHAR (36)
+          //when testing need to see what datatype sequelize actually assigns
+          type: DataTypes.UUID,
+          primaryKey: true,
+          defaultValue: DataTypes.UUIDV4,
+          allowNull:false
+      },
+      user_id:
+      {
+        type: DataTypes.UUID,
+        allowNull: false
+      },
+      age:
+      {
+          type: DataTypes.INTEGER,
+          validate:
+          {
+              len: [1,2]
+          },
+          allowNull: true
+      },
+      gender:
+      {
+          type: DataTypes.ENUM,
+          values: ['Male', 'Female', 'Trans-gender', 'Non-Binary'],
+          allowNull: true
+      },
+      phone:
+      {
+          //would like front end validation for this value
+          //need to decide how to store, likely need to normalize
+          //thinking (xxx) xxx-xxxx
+          type: DataTypes.STRING,
+          validate:
+          {
+              len: [10,20]
+          },
+          allowNull: true
+      },
+      city:
+      {
+          type: DataTypes.UUID,
+          allowNull: true
+      },
+      state:
+      {
+          type: DataTypes.UUID,
+          allowNull: true
+      },
+      country:
+      {
+          type: DataTypes.UUID,
+          allowNull: true
+      },
+      github:
+      {
+          type: DataTypes.STRING,
+          allowNull: true
+      },
+      linkedin:
+      {
+          type: DataTypes.STRING,
+          allowNull: true
+      },
+      avatar:
+      {
+          type: DataTypes.BLOB,
+          allowNull: true
+      },
+      //need query place averaged value of entries from reputation into this field
```

Comment:
How will reputation work? I am just curious for academic reasons.

Reply:
this field is purely for future implementation, based on the agreed upon desire to have a review of user process.  there is currently no work flow for it and it is not a required field in the database.  

conceptually, my thought was that when someone reviews you (numeric scale, capturing both reviewer and reviewee) then we could run a stored procedure to take the weighted average of all reviews and populate that field.